added vertex color picking from fbxloader2

jonobr1 opened the pull request:
Updated and fixed patch of this original PR: https://github.com/mrdoob/three.js/pull/10780#pullrequestreview-21132542

jonobr1 committed:
added vertex color picking from fbxloader2

mrdoob commented:
@Kyle-Larson Looks good?

jonobr1 commented:
one thing I wasn't sure about was how to let the mesh's material know that there were vertex colors. In my demos once I load in the model I will set the `mesh.material.vertexColors = THREE.VertexColors;`. I wasn't sure if the `FBXLoader2` should do that out of the box for you or not..?

Kyle-Larson commented:
@jonobr1 Do you happen to have an example fbx file with vertex colors that I could look at?  Might help in determining how to handle setting the vertex colors on the material for the mesh.

jonobr1 commented:
Yes, I'll get one cleared and throw it up online for you to check out.

jonobr1 commented:
@Kyle-Larson: https://dl.dropboxusercontent.com/u/1595444/issues/three.js/fbx-loader-example/index.html

Kyle-Larson commented:
So as far as I see, there's nothing in the material that really says how to color the model if vertex colors also exist on the geometry.  We could make the assumption that if vertex colors exist on the geometry, that all instances of that geometry should use vertex colors, although I'm not 100% sure if that is how fbx assumes to use vertex colors in the shader.  I say we just make a simple check when generating a mesh that says if the buffer geometry has a colors attribute, then set the materials for that instance to `useVertexColors` and if someone runs into an issue with it, then hopefully they bring up an issue and we can go forward from there.

jonobr1 commented:
I'll add that right now!

Commit:
added check to enable VertexColors in the shader if detected

jonobr1 commented:
This latest commit has that change and I also updated the dropbox example. Because Maya's default exported material is `Lambert` I added a light to show that the donut is still be added to the scene correctly. https://dl.dropboxusercontent.com/u/1595444/issues/three.js/fbx-loader-example/index.html

mrdoob commented:
Thanks!

mrdoob commented:
@Kyle-Larson 

I have change it so it checks for `LayerElementColor` before creating the `color` attribute. https://github.com/mrdoob/three.js/commit/c2308789d88c8e281f21a52dc809a674a0aaa1dc

I did this because checking for `bufferInfo.colorBuffer.length > 0` returns `true` as we're creating a buffer full of zeros. Do you think we should do the same for `bufferInfo.normalBuffer.length > 0` and `bufferInfo.uvBuffer.length > 0`?

Kyle-Larson commented on examples/js/loaders/FBXLoader2.js:
```diff
@@ -1212,6 +1212,11 @@
 								}
+								if ( 'color' in geometry.attributes && geometry.attributes.color.count > 0 ) {
+
+									material.vertexColors = THREE.VertexColors;
```
You might need to iterate through the array of materials here instead of setting the vertexColors property on the Multi-Material (If there is a multi-material).  The code here admittedly is a little inconsistent, I might go ahead and make a quick fix.

Kyle-Larson replied:
Fixed in ( #10790 )
